zetaclient/zetacore_observer.go

```diff
@@ -172,7 +172,10 @@ func (co *CoreObserver) startSendScheduler() {
 							outTxID := fmt.Sprintf("%s-%d-%d", cctx.Index, params.ReceiverChainId, nonce) // would outTxID a better ID?
 
 							// Process Bitcoin OutTx
-							if common.IsBitcoinChain(c.ChainId) && !outTxMan.IsOutTxActive(outTxID) {
+							if common.IsBitcoinChain(c.ChainId) {
+								if outTxMan.IsOutTxActive(outTxID) {
+									break // there is no need to process cctx with future nonces
```

Comment:
I think we should make the comment more descriptive. Why `IsOutTxActive == true` implies there is future nonces, why there is no need to process cctx with future nonces